src/tribler/core/components/watch_folder/watch_folder.py

```diff
@@ -64,23 +66,37 @@ def check_watch_folder(self):
         self._logger.debug(f'Watch dir: {watch_dir}')
 
         for root, _, files in os.walk(watch_dir):
-            root = path_util.Path(root)
             for name in files:
                 if not name.endswith(".torrent"):
                     continue
 
+                path = Path(root) / name
+                self._logger.info(f'Torrent file found: {path}')
+                if path in self.skipped_torrents:
+                    self._logger.info(f'Skip: {path}')
+                    continue
+
                 try:
-                    tdef = TorrentDef.load(root / name)
+                    tdef = TorrentDef.load(path)
                     if not tdef.get_metainfo():
-                        self.cleanup_torrent_file(root, name)
+                        self.cleanup_torrent_file(path)
                         continue
-                except:  # torrent appears to be corrupt
-                    self.cleanup_torrent_file(root, name)
+                except Exception as e:  # pylint: disable=broad-except
+                    self._logger.warning(f'Torrent appears to be corrupt: {e}')
+                    self.cleanup_torrent_file(path)
                     continue
 
-                infohash = tdef.get_infohash()
+                try:
+                    infohash = tdef.get_infohash()
+
+                    if not self.download_manager.download_exists(infohash):
+                        self._logger.info("Starting download from torrent file %s", name)
+                        download_config = DownloadConfig.convert(self.download_manager.download_defaults)
+                        download_config.state_dir = self.state_dir
+                        self.download_manager.start_download(torrent_file=path, config=download_config)
+                except Exception as e:  # pylint: disable=broad-except
```

Comment:
This is a new broad try/except block, and I think we should discuss if it is necessary to have it. If yes, we should comment on its purpose for new developers. Every overly broad try/except block brings a danger of accidentally catching and suppressing some unrelated error that will remain unnoticed and unfixed because of that. Having as narrow and specific try/except block as possible is better.

Looking at the PR and the original issue, it is unclear what type of exceptions this try/except block should catch. I see several tracebacks listed in the issue:

1. [PermissionError](https://github.com/Tribler/tribler/issues/5833#issuecomment-1139377949) when calling `Path.is_dir()` in the [previous version](https://github.com/Tribler/tribler/blob/20daa03c739f789bae1d4f1bdea0b233aef0d868/src/tribler-core/tribler_core/components/watch_folder/watch_folder.py#L47) of the `check_watch_folder` method (in release 7.11):

```python
    def check_watch_folder(self):
        if not self.watch_folder.is_dir():
            return
        ...
```

In #6863, the logic of the `check_watch_folder` method has been rewritten, but it seems to me that this problem is still here, as line 60 of the current code can still raise `PermissionError`:
```python
if not directory.is_dir():
```
Probably it should be wrapped with `except OSError: ...`.

2. [Exception in `process_alert`](https://github.com/Tribler/tribler/issues/5833#issue-761191737). The `process_alert_handler` is called from the asynchronous [`process_alert` task](https://github.com/Tribler/tribler/blob/1609e8d85dc42c87db8fbe640404877ee1d2edf6/src/tribler/core/components/libtorrent/download_manager/download_manager.py#L153).

Neither of these two exceptions is thrown from the `download_manager.start_download` method. So, this broad try/except block can catch neither of these errors. Because of this, I don't see the purpose of this try/except block. Maybe it is possible to remove it.

Reply:
If the `process_alert` call throws an exception, the torrent will not be added to the `skipped_torrents` set. It looks like torrents should be added to the `skipped_torrents` set inside the `process_alert` call and not here.

Reply:
Fixed